### PR TITLE
Add plugins support

### PR DESCRIPTION
**Pull request by robertbaldyga**
Signed-off-by: Robert Baldyga <robert.baldyga@intel.com>

### CODE REVIEW

File: internal_plugins/example_plugin/__init__.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#
+# Copyright(c) 2019 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+#
+
+
+class ExamplePlugin:
+    def __init__(self, params):
+        print("Example plugin initialized")
```
Comment (Deixx) [inline internal_plugins/example_plugin/__init__.py]: you may add 'self.params = params' for clarity

File: core/test_run_utils.py [hunk 2]
```diff
@@ -35,8 +40,15 @@ def __configure(cls, config):
...
 @classmethod
-def __prepare(cls, item):
+def __prepare(cls, item, config):
+    if not config:
+        raise Exception("You need to specify DUT config!")
+
     cls.item = item
+    cls.config = config
+    cls.req_plugins = {}
+    cls.opt_plugins = {}
```
Comment (Deixx) [inline core/test_run_utils.py]: Looks like those are not needed

File: internal_plugins/example_plugin/__init__.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
+#
+# Copyright(c) 2019 Intel Corporation
```
Comment (Deixx) [inline internal_plugins/example_plugin/__init__.py]: 2020 ;)

File: core/test_run_utils.py [hunk 4]
```diff
@@ -25,6 +26,10 @@ def __configure(cls, config):
         "markers",
         "require_disk(name, type): require disk of specific type, otherwise skip"
     )
+    config.addinivalue_line(
+        "markers",
+        "require_plugin(name, params): require specific plugins, otherwise skip"
```
[the remark this answers is not in the document]
Comment (robertbaldyga) [reply core/test_run_utils.py]: Done.

File: internal_plugins/example_plugin/__init__.py (hunk 1 above)
Comment (robertbaldyga) [reply internal_plugins/example_plugin/__init__.py]: Added.

File: core/test_run_utils.py (hunk 2 above)
Comment (robertbaldyga) [reply core/test_run_utils.py]: Yep, removed.

File: internal_plugins/example_plugin/__init__.py (hunk 3 above)
Comment (robertbaldyga) [reply internal_plugins/example_plugin/__init__.py]: Oh no, we also need to update it everywhere else in the code again. Why these years are flying by so quickly?

**Pull request merged**